### DISCUSSION TITLE
Segmentation fault - invalid memory reference

### DISCUSSION THREAD

**Discussion by jk262626** [nakib/elphbolt#156] [Q&A] [answered]
Dear Nakib,

I have been using Elphbolt on the shared cluster system, but I recently switched back to my local workstation due to resource limitations. My workstation runs Ubuntu 23.10 and is equipped with an Intel processor with 128 cores and 128GB of memory. After installing Elphbolt and running the test for SiC, I encountered a segmentation fault, as shown in the screenshot below:
![Screenshot from 2024-10-22 18-41-01](https://github.com/user-attachments/assets/bef53aac-28d0-4745-b94c-690a4e74912d)
I suspected the issue was related to multi-core usage, and when I changed the cafrun setting from 2 to 1, the problem was resolved. Additionally, when I monitored memory usage during the calculations with 2 cores, I did not observe any memory overflow issues. Interestingly, I never experienced this issue while using Elphbolt on the cluster system.

Do you have any recommendations for further troubleshooting or adjustments I could try?

Thank you!

**Comment by nakib** [comment]
Dear Jiyoung,

Thanks for reporting this. This is likely due to your use of gcc13. Please see the solution given here: https://github.com/nakib/elphbolt/discussions/155

**Reply by jk262626** [reply, marked as answer]
Thank you Nakib!

I resolved problem following your comments. 
I have been using Spack to install Opencoarrays. 
I used this command (`spack install opencoarrays`) before and it didn't work because default gcc for spack was 13.2.0 version.
This time I changed the command to `spack install opencoarrays@2.10.1 ^mpich %gcc@12.3.0` to specify gcc version to be used for opencoarrays and the issue was resolved.

**Reply by nakib** [reply]
Thank you for reporting back that you found a solution.

Best,
Nakib
